File: port_java_OBSOLUTE/com/boare/cadencii/AttachedCurve.java
/*
 * AttachedCurve.cs
 * Copyright (c) 2008-2009 kbinani
 *
 * This file is part of Boare.Cadencii.
 *
 * Boare.Cadencii is free software; you can redistribute it and/or
 * modify it under the terms of the GPLv3 License.
 *
 * Boare.Cadencii is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
package com.boare.cadencii;

import java.util.*;

public class AttachedCurve implements Cloneable {
    public Vector<BezierCurves> curves = new Vector<BezierCurves>();

    public static String getXmlElementName( String name ){
        if( name.equals( "curves" ) ){
            return "Curves";
        }
        return name;
    }

    public static String getGenericTypeName( String name ){
        if( name.equals( "curves" ) ){
            return "com.boare.cadencii.BezierCurves";
        }
        return "";
    }

    public void add( BezierCurves item ) {
        curves.add( item );
    }

    public void removeElementAt( int index ) {
        curves.removeElementAt( index );
    }

    public void insert( int position, BezierCurves attached_curve ) {
        curves.insertElementAt( attached_curve, position );
    }

    public Object clone() {
        AttachedCurve ret = new AttachedCurve();
        ret.curves.clear();
        for ( int i = 0; i < curves.size(); i++ ) {
            ret.curves.add( (BezierCurves)curves.get( i ).clone() );
        }
        return ret;
    }

    public BezierCurves get( int index ){
        return curves.get( index );
    }

    public void set( int index, BezierCurves value ){
        curves.set( index, value );
    }
}

File: src/Cadencii/AttachedCurve.cs
/*
 * AttachedCurve.cs
 * Copyright © 2008-2011 kbinani
 *
 * This file is part of cadencii.
 *
 * cadencii is free software; you can redistribute it and/or
 * modify it under the terms of the GPLv3 License.
 *
 * cadencii is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 */
using System;
using System.Collections.Generic;
using cadencii;
using cadencii.java.util;
using cadencii.java.io;

namespace cadencii
{

    public class AttachedCurve : ICloneable
    {
        private List<BezierCurves> mCurves = new List<BezierCurves>();

        /// <summary>
        /// XML保存用
        /// </summary>
        public List<BezierCurves> Curves
        {
            get
            {
                return getCurves();
            }
            set
            {
                setCurves(value);
            }
        }

        public List<BezierCurves> getCurves()
        {
            return mCurves;
        }

        public void setCurves(List<BezierCurves> value)
        {
            mCurves = value;
        }

        public BezierCurves get(int index)
        {
            return mCurves[index];
        }

        public void set(int index, BezierCurves value)
        {
            mCurves[index] = value;
        }

        public void add(BezierCurves item)
        {
            mCurves.Add(item);
        }

        public void removeElementAt(int index)
        {
            mCurves.RemoveAt(index);
        }

        public void insertElementAt(int position, BezierCurves attached_curve)
        {
            mCurves.Insert(position, attached_curve);
        }

        public Object clone()
        {
            AttachedCurve ret = new AttachedCurve();
            ret.mCurves.Clear();
            int c = mCurves.Count;
            for (int i = 0; i < c; i++) {
                ret.mCurves.Add((BezierCurves)mCurves[i].clone());
            }
            return ret;
        }

        public Object Clone()
        {
            return clone();
        }
    }

}
